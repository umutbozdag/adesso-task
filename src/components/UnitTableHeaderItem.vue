<template>
  <th class="px-6 py-3">
    {{ tableHeader.label }}
  </th>
</template>

<script setup lang="ts">
import { IUnitTableHeaderItem } from '@/types/UnitTable';

interface IUnitTableHeaderItemProps {
  tableHeader: IUnitTableHeaderItem;
}

defineProps<IUnitTableHeaderItemProps>();
</script>
