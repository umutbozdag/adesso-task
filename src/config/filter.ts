import { UnitAgeEnum } from '@/types/Unit';
import { FilterDropdownOption } from '@/types/FilterDropdown';

export const filterOptionsConfig: FilterDropdownOption[] = [
  {
    label: UnitAgeEnum.All,
    name: UnitAgeEnum.All
  },
  {
    label: UnitAgeEnum.Dark,
    name: UnitAgeEnum.Dark
  },
  {
    label: UnitAgeEnum.Feudal,
    name: UnitAgeEnum.Feudal
  },
  {
    label: UnitAgeEnum.Castle,
    name: UnitAgeEnum.Castle
  },
  {
    label: UnitAgeEnum.Imperial,
    name: UnitAgeEnum.Imperial
  }
];
