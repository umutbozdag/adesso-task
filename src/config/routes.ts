import { RouteConfigItem, RoutesNameEnum, RoutesPathEnum } from '@/types/Route';

const routes: Record<RoutesNameEnum, RouteConfigItem> = {
  [RoutesNameEnum.Home]: {
    path: RoutesPathEnum.Home,
    name: RoutesNameEnum.Home,
    showInNavbar: true,
    label: 'Home'
  },
  [RoutesNameEnum.Units]: {
    path: RoutesPathEnum.Units,
    name: RoutesNameEnum.Units,
    showInNavbar: true,
    label: 'Units'
  },
  [RoutesNameEnum.UnitDetails]: {
    path: RoutesPathEnum.UnitDetails,
    name: RoutesNameEnum.UnitDetails
  },
  [RoutesNameEnum.NotFound]: {
    path: RoutesPathEnum.NotFound,
    name: RoutesNameEnum.NotFound
  }
};

export default routes;
