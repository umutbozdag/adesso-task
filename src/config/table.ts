import { IUnitTableHeaderItem } from '@/types/UnitTable';

export function createTableHeaders(): IUnitTableHeaderItem[] {
  return [
    {
      id: 'id',
      label: 'ID'
    },
    {
      id: 'name',
      label: 'Name'
    },
    {
      id: 'age',
      label: 'Age'
    },
    {
      id: 'costs',
      label: 'Costs'
    }
  ];
}
