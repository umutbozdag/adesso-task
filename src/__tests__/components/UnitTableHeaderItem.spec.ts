import UnitTableHeaderItem from '@/components/UnitTableHeaderItem.vue';
import { VueWrapper, mount } from '@vue/test-utils';
import { expect, it, describe, beforeAll } from 'vitest';
import { ComponentPublicInstance } from 'vue';
import { createTableHeaders } from '@/config/table';

describe('UnitTableFilters', () => {
  let wrapper: VueWrapper<ComponentPublicInstance & { [key: string]: any }>;
  beforeAll(() => {
    wrapper = mount(UnitTableHeaderItem, {
      props: {
        tableHeader: createTableHeaders()[0]
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows label correct', async () => {
    expect(wrapper.html()).toContain('ID');
  });
});
