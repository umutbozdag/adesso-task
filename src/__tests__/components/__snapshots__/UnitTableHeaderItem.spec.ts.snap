// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnitTableFilters > renders correctly 1`] = `"<th class=\\"px-6 py-3\\">ID</th>"`;
